flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/filemerging/FileMergingSnapshotManagerBase.java

```diff
@@ -607,6 +619,7 @@ public void restoreStateHandles(
                                                                     fileHandle.getScope()))
                                                             ? physicalFileDeleter
                                                             : null;
+                                            spaceStat.onPhysicalFileCreate();
```

Comment:
I'm afraid this is not true. If the `isManagedByFileMergingManager` returns false, there will be no file deleter, so no `spaceStat` will be updated after file deletion. I'd suggest skip `spaceStat` if `isManagedByFileMergingManager` returns false. WDYT?

Reply:
Agree👍, only update `spaceStat` when `isManagedByFileMergingManager` return true.